### ISSUE TITLE
bug:  github workflow `Run rpc tests/rpc-tests` cannot get InvokeTransactionReceipt

### ISSUE THREAD

**Issue by fishseabowl** [keep-starknet-strange/madara#1557]
# Bug Report

**Madara version:**

v.0.7.0

**Current behavior:**

In the PR #1526 , github workflow [Run rpc tests/rpc-tests](https://github.com/keep-starknet-strange/madara/actions/runs/8594755121/job/23548606964?pr=1526), the test case starknet-rpc-test::tarknet_estimate_fee::work_ok cannot get InvokeTransactionReceipt, but locally, the test case is ok.

**Expected behavior:**

The function JsonRpcClient.get_transaction_receipt can return InvokeTransactionReceipt

**Steps to reproduce:**

github workflow [Run rpc tests/rpc-tests](https://github.com/keep-starknet-strange/madara/actions/runs/8594755121/job/23548606964?pr=1526)

**Related code:**

``` rust
loop {
        let invoke_tx_receipt = rpc.get_transaction_receipt(invoke_transaction_result.transaction_hash).await?;

        match invoke_tx_receipt {
            MaybePendingTransactionReceipt::Receipt(TransactionReceipt::Invoke(receipt)) => {
                assert_eq!(FieldElement::from(estimates[0].overall_fee), receipt.actual_fee);
                break; // Break the loop if receipt is received
            }

            MaybePendingTransactionReceipt::PendingReceipt(PendingTransactionReceipt::Invoke(_)) => {
                time::sleep(Duration::from_secs(SLEEP_DURATION)).await;
            }

            _ => {
                panic!("expected invoke transaction receipt");
            }
        }
    }
```

**Other information:**

**Comment by tdelabro** [comment]
@fishseabowl I'm not sure to understand what is the problem here?
And is it still there in the latest state of the codebase?

**Comment by tdelabro** [comment]
@fishseabowl is it still a thing?

**Comment by fishseabowl** [comment]
@tdelabro I think this issue has been resolved in PR #1601
